reporter/src/main/kotlin/reporters/CycloneDxReporter.kt

```diff
@@ -114,23 +117,27 @@ class CycloneDxReporter : Reporter {
         ortResult.getPackages().forEach { (pkg, _) ->
             // TODO: We should actually use the concluded license expression here, but we first need a workflow to
             //       ensure it is being set.
-            val licenseNames = pkg.declaredLicensesProcessed.allLicenses +
-                    ortResult.getDetectedLicensesForId(pkg.id, input.packageConfigurationProvider)
-
-            val licenseObjects = licenseNames.map { licenseName ->
-                val spdxId = SpdxLicense.forId(licenseName)?.id
-
-                // Prefer to set the id in case of an SPDX "core" license and only use the name as a fallback, also
-                // see https://github.com/CycloneDX/cyclonedx-core-java/issues/8.
-                License().apply {
-                    id = spdxId
-                    name = licenseName.takeIf { spdxId == null }
-                    licenseText = LicenseText().apply {
-                        contentType = "plain/text"
-                        text = input.licenseTextProvider.getLicenseText(licenseName)
-                    }
-                }
+            val declaredLicenseNames = pkg.declaredLicensesProcessed.allLicenses.toSortedSet()
+            val detectedLicenseNames = ortResult.getDetectedLicensesForId(pkg.id, input.packageConfigurationProvider)
+
+            fun mapLicenseNamesToObjects(licenseNames: SortedSet<String>, origin: String): List<License> {
```

Comment:
Let's move this as a private function to the class level for a better overview. Also, `licenseNames` can be reduced to type `Collection<String>` so that `declaredLicenseNames` do esnot need to be converted to a `SortedSet` above.